src/main/java/Asuka.java

```diff
@@ -0,0 +1,256 @@
+import java.util.Scanner;
+
+class Task {
+    protected String description;
+    protected boolean isDone;
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public void markAsDone() {
+        this.isDone = true;
+    }
+
+    public void markAsUndone() {
+        this.isDone = false;
+    }
+
+    public void printTask() {
+        if (this.isDone) {
+            System.out.println("[X] " + this.description);
+        } else {
+            System.out.println("[ ] " + this.description);
+        }
+    }
+}
+
+class Todo extends Task {
+
+    public Todo(String description) {
+        super(description);
+    }
+
+    public void printTask() {
+        if (this.isDone) {
+            System.out.println("[T][X] " + this.description);
+        } else {
+            System.out.println("[T][ ] " + this.description);
+        }
+    }
+}
+
+class Deadline extends Task {
+
+    public Deadline(String description) {
+        super(description);
+    }
+
+    public void printTask() {
+        if (this.isDone) {
+            System.out.println("[D][X] " + this.description);
+        } else {
+            System.out.println("[D][ ] " + this.description);
+        }
+    }
```

Comment:
Well done on incorporating method a printTask() into the Task class itself.
A fine display of encapsulation in OOP.